contracts/script/LightClientArb.s.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import { Script } from "forge-std/Script.sol";
+import { LightClientArbitrum } from "../src/LightClientArbitrum.sol";
+import { LightClient as LC } from "../src/LightClient.sol";
+import { Upgrades } from "openzeppelin-foundry-upgrades/Upgrades.sol";
+
+contract DeployLightClientArbitrum is Script {
+    // Deployment Errors
+    error SetPermissionedProverFailed();
+    error OwnerTransferFailed();
+    error RetentionPeriodIsNotSetCorrectly();
+    error InitialStateIsNotSetCorrectly();
+
+    function run(uint32 numInitValidators, uint32 stateHistoryRetentionPeriod, address owner)
+        public
+        returns (
+            address proxyAddress,
+            address implementationAddress,
+            LC.LightClientState memory lightClientState
+        )
+    {
+        address deployer;
+
+        string memory ledgerCommand = vm.envString("USE_HARDWARE_WALLET");
+        if (keccak256(bytes(ledgerCommand)) == keccak256(bytes("true"))) {
+            deployer = vm.envAddress("DEPLOYER_HARDWARE_WALLET_ADDRESS");
+        } else {
+            // get the deployer info from the environment
+            string memory seedPhrase = vm.envString("DEPLOYER_MNEMONIC");
+            uint32 seedPhraseOffset = uint32(vm.envUint("DEPLOYER_MNEMONIC_OFFSET"));
+            (deployer,) = deriveRememberKey(seedPhrase, seedPhraseOffset);
+        }
+
+        vm.startBroadcast(deployer);
+
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "mock-genesis";
+        cmds[2] = vm.toString(uint256(numInitValidators));
+
+        bytes memory result = vm.ffi(cmds);
+        (LC.LightClientState memory state, LC.StakeTableState memory stakeState) =
+            abi.decode(result, (LC.LightClientState, LC.StakeTableState));
+
+        proxyAddress = Upgrades.deployUUPSProxy(
```

Comment:
why is this a proxy address? why cant we just deploy an implementation contract and then upgrade our existing proxy?

Reply:
opted for a deploy instead of an upgrade whilst dealing with the precompile issue. once this quick review is fine, i can continue with the upgrade steps

Reply:
created a ticket for the upgrade script as the safe transaction service is not supported in arbitrum sepolia. this service is needed for proposing transactions to the safe multisig wallet, created another ticket https://github.com/EspressoSystems/espresso-sequencer/issues/2457